starter-code/src/functions-and-arrays.js

```diff
@@ -1,100 +1,438 @@
 // Find the maximum
 
+function maxOfTwoNumbers(a, b) {
+  if (a > b) {
+    return a;
+  } else {
+    return b;
+  }
+}
+m;
+console.log(maxOfTwoNumbers(4, 2));
 // Finding Longest Word
+
 var words = [
-  'mystery',
-  'brother',
-  'aviator',
-  'crocodile',
-  'pearl',
-  'orchard',
-  'crackpot'
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
 ];
+var lgth = 0;
+
+for (var index = 0; index < words.length; index++) {
+  if (words[index].length > lgth) {
+    var lgth = words[index].length;
+    answer = words[index];
+  }
+}
+
+console.log(answer);
+
+function findLongestWord(words) {
+  words.forEach(function(wordsIndex) {
+    if (wordsIndex.length > lght) {
+      var lght = wordsIndex.length;
+      answer = wordsIndex;
+    }
+  });
+}
+console.log(answer);
 
 // Calculating a Sum
 
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+var sum = 0;
+function sumArray(numbers) {
+  for (var index = 0; index < numbers.length; index++) {
+    sum = sum + numbers[index];
+  }
+  console.log(sum);
+}
+sumArray(numbers);
+
 // Calculate the Average
 
 var numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+var sum1 = 0;
+var average = 0;
+
+function sumArray(numbers) {
+  for (var index = 0; index < numbersAvg.length; index++) {
+    sum1 = sum1 + numbersAvg[index];
+  }
+  console.log(sum1);
+  console.log((average = sum1 / numbers.length));
+}
+sumArray(umbersAvg);
+
+// second possibility using the previous function sum
+var sum1 = 0;
+function sumArray(numbers) {
+  for (var index = 0; index < numbers.length; index++) {
+    sum1 = sum1 + numbers[index];
+  }
+  return sum1;
+}
+
+function averageNumbers(fff) {
+  var average = sumArray(fff) / fff.length;
+  return average;
+}
+averageNumbers(numbersAvg);
 
```

Comment:
If you put twice the same function, Jasmine is not gonna like it. :)

The first part is really confused but you did well in the second part. Just the variable is still declared outside the function and should be inside and the condition that if the array is empty is not written but it's right.

Here is how you could have it completely right: 
```
function sumArray(numbers) {
  var sum1 = 0;
  for (var index = 0; index < numbers.length; index++) {
    sum1 = sum1 + numbers[index];
  }
  return sum1;
}

function averageNumbers(fff) {
  if (fff.length === 0) {
    return undefined;
  }
  var average = sumArray(fff) / fff.length;
  return average;
}
averageNumbers(numbersAvg);
```